### PR TITLE
Fix resolving paths for .face.icon in dashboard and add home ('~') path support

### PR DESCRIPTION
**Pull request by davfsa**
I noticed that my `.face.icon` was not being loaded and, after some investigation, I uncovered that `GdkPixbuf.Pixbuf` doesn't expand paths, let alone environment variables (which the default config was attempting to use through `$HOME`).

This fixes it by adding support for `~` to `isAnImage`, making use of it in avatar resolution and resolving the path to load the avatar.

---

I have also simplified `isAnImage` by removing the double check for file existence, as `GdkPixbuf.Pixbuf.new_from_file` does it itself

### CODE REVIEW

File: src/lib/utils.ts [hunk 1]
```diff
@@ -228,25 +227,39 @@ export function launchApp(app: AstalApps.Application): void {
...
+ * Note: This will only expand '~' if present. Path traversal is not supported.
+ *
+ * @param path The path to resolve.
+ *
+ * @returns The absolute representation of the resolved path.
+ */
+export function resolvePath(path: string): string {
+    if (path.charAt(0) == '~') {
+        path = GLib.get_home_dir() + '/' + path.substring(1, path.length);
+    }
```
Comment (davfsa) [inline src/lib/utils.ts:258]:
```suggestion
    return path.replace('~', GLib.get_home_dir())
```

Might be a bit better

File: src/lib/utils.ts [hunk 2]
```diff
@@ -228,25 +227,37 @@ export function launchApp(app: AstalApps.Application): void {
...
+ * Note: This will only expand '~' if present. Path traversal is not supported.
+ *
+ * @param path The path to resolve.
+ *
+ * @returns The absolute representation of the resolved path.
+ */
+export function resolvePath(path: string): string {
+    return path.replace('~', GLib.get_home_dir())
+
+    return path;
```
Comment (Jas-SinghFSU) [inline src/lib/utils.ts:258]:
```suggestion
```

File: src/lib/utils.ts [hunk 3]
```diff
@@ -228,25 +227,35 @@ export function launchApp(app: AstalApps.Application): void {
...
+/**
+ * Resolve a path to the absolute representation of the path.
+ *
+ * Note: This will only expand '~' if present. Path traversal is not supported.
+ *
+ * @param path The path to resolve.
+ *
+ * @returns The absolute representation of the resolved path.
+ */
+export function resolvePath(path: string): string {
```
Comment (Jas-SinghFSU) [inline src/lib/utils.ts:255]: We should probably name this `normalizePath` as it simply makes it absolute and doesn't quite resolve a path.

Comment (davfsa) [reply src/lib/utils.ts:255]: Sounds good!

**Comment by Jas-SinghFSU** [comment]
Thanks! :smile:

**Pull request merged**